Tests/SentryTests/Integrations/Screenshot/SentryScreenshotIntegrationTests.swift

```diff
@@ -171,6 +171,29 @@ class SentryScreenshotIntegrationTests: XCTestCase {
         
     }
     
+    func test_backgroundForAppHangs() {
+        let sut = fixture.getSut()
+        let testVH = TestSentryScreenshot()
+        SentryDependencyContainer.sharedInstance().screenshot = testVH
+        
+        let event = Event()
+        event.exceptions = [Sentry.Exception(value: "test", type: "App Hanging")]
+
+        let ex = expectation(description: "Attachment Added")
+        
+        testVH.processViewHierarchyCallback = {
+            ex.fulfill()
+            XCTAssertFalse(Thread.isMainThread)
+        }
+        
+        let dispatch = DispatchQueue(label: "background")
+        dispatch.async {
```

Comment:
`l`: A bit shorter if you want:
```suggestion
        DispatchQueue.global().async {
```